src/components/ContextBar.test.js

```diff
@@ -0,0 +1,625 @@
+import { tools } from 'src/pages/workspaces/workspace/analysis/notebook-utils'
+import { getGalaxyComputeCost, getGalaxyDiskCost, getPersistentDiskCostHourly, getRuntimeCost, runtimeConfigCost } from 'src/pages/workspaces/workspace/analysis/runtime-utils'
+
+
+const GALAXY_COMPUTE_COST = 10
+const GALAXY_DISK_COST = 1
+const RUNTIME_COST = 0.1
+const PERSISTENT_DISK_COST = 0.01
+
+jest.mock('src/pages/workspaces/workspace/analysis/runtime-utils', () => {
+  const originalModule = jest.requireActual('src/pages/workspaces/workspace/analysis/runtime-utils')
+
+  return {
+    ...originalModule,
+    getGalaxyComputeCost: jest.fn(),
+    getGalaxyDiskCost: jest.fn(),
+    getPersistentDiskCostHourly: jest.fn(),
+    getRuntimeCost: jest.fn(),
+    runtimeConfigCost: jest.fn()
+  }
+})
+
+// Mocking for terminalLaunchLink using Nav.getLink
+jest.mock('src/libs/nav', () => {
+  const originalModule = jest.requireActual('src/libs/nav')
+
+  return {
+    ...originalModule,
+    getPath: jest.fn(() => '/test/'),
+    getLink: jest.fn(() => '/')
+  }
+})
+
+// Mocking PopupTrigger to avoid test environment issues with React Portal's requirement to use
+// DOM measure services which are not available in jest environment
+jest.mock('src/components/PopupTrigger', () => {
+  const originalModule = jest.requireActual('src/components/PopupTrigger')
+  return {
+    ...originalModule,
+    MenuTrigger: jest.fn()
+  }
+})
+
+jest.mock('src/pages/workspaces/workspace/analysis/modals/CloudEnvironmentModal', () => {
+  const originalModule = jest.requireActual('src/pages/workspaces/workspace/analysis/modals/CloudEnvironmentModal')
+  return {
+    ...originalModule,
+    CloudEnvironmentModal: jest.fn()
+  }
+})
+
+jest.mock('src/libs/config', () => {
+  const originalModule = jest.requireActual('src/libs/config')
+  return {
+    ...originalModule,
+    isCromwellAppVisible: () => {
+      return true
+    }
+  }
+})
+
+// Necessary to mock the AJAX module.
+jest.mock('src/libs/ajax')
+const mockRuntimesStartFn = jest.fn()
+const mockRuntime = jest.fn()
+
+beforeEach(() => {
+  MenuTrigger.mockImplementation(({ content }) => { return div([content]) })
+  CloudEnvironmentModal.mockImplementation(({ isOpen, filterForTool, onSuccess, onDismiss, ...props }) => {
+    return isOpen ? div([
+      'Cloud Environment Details',
+      div(filterForTool),
+      div({ label: 'Success Button', onClick: () => onSuccess() }, 'SuccessButton'),
+      div({ label: 'Success Button', onClick: () => onDismiss() }, 'DismissButton')
+    ]) : div([])
+  })
+
+  Ajax.mockImplementation(() => {
+    mockRuntime.mockReturnValue({ start: mockRuntimesStartFn })
+    return {
+      Metrics: {
+        captureEvent: () => {}
+      },
+      Runtimes: {
+        runtime: mockRuntime
+      }
+    }
+  })
+
+  getGalaxyComputeCost.mockImplementation(() => {
+    return GALAXY_COMPUTE_COST
+  })
```

Comment:
These could all use `mockReturnValue` instead of `mockImplementation`.

```suggestion
  getGalaxyComputeCost.mockReturnValue(GALAXY_COMPUTE_COST)
```